clients/ui/frontend/src/app/pages/modelRegistry/screens/RegisterModel/utils.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  ModelArtifact,
+  ModelArtifactState,
+  ModelState,
+  ModelVersion,
+  RegisteredModel,
+} from '~/app/types';
+import { ModelRegistryAPIState } from '~/app/context/useModelRegistryAPIState';
+import { objectStorageFieldsToUri } from '~/app/pages/modelRegistry/screens/utils';
+import {
+  ModelLocationType,
+  RegisterModelFormData,
+  RegisterVersionFormData,
+  RegistrationCommonFormData,
+} from './useRegisterModelData';
+
+export type RegisterModelCreatedResources = RegisterVersionCreatedResources & {
+  registeredModel: RegisteredModel;
+};
+
+export type RegisterVersionCreatedResources = {
+  modelVersion: ModelVersion;
+  modelArtifact: ModelArtifact;
+};
+
+export const registerModel = async (
+  apiState: ModelRegistryAPIState,
+  formData: RegisterModelFormData,
+  author: string,
+): Promise<RegisterModelCreatedResources> => {
+  const registeredModel = await apiState.api.createRegisteredModel(
+    {},
+    {
+      name: formData.modelName,
+      description: formData.modelDescription,
+      customProperties: {},
+      owner: author,
+      state: ModelState.LIVE,
+    },
+  );
+  const { modelVersion, modelArtifact } = await registerVersion(
+    apiState,
+    registeredModel,
+    formData,
+    author,
+  );
+  return { registeredModel, modelVersion, modelArtifact };
+};
+
+export const registerVersion = async (
+  apiState: ModelRegistryAPIState,
+  registeredModel: RegisteredModel,
+  formData: Omit<RegisterVersionFormData, 'registeredModelId'>,
+  author: string,
+): Promise<RegisterVersionCreatedResources> => {
+  const modelVersion = await apiState.api.createModelVersionForRegisteredModel(
+    {},
+    registeredModel.id,
+    {
+      name: formData.versionName,
+      description: formData.versionDescription,
+      customProperties: {},
+      state: ModelState.LIVE,
+      author,
+      registeredModelId: registeredModel.id,
+    },
+  );
+  const modelArtifact = await apiState.api.createModelArtifactForModelVersion({}, modelVersion.id, {
+    name: `${registeredModel.name}-${formData.versionName}-artifact`,
+    description: formData.versionDescription,
+    customProperties: {},
+    state: ModelArtifactState.LIVE,
+    author,
+    modelFormatName: formData.sourceModelFormat,
+    modelFormatVersion: formData.sourceModelFormatVersion,
+    // TODO fill in the name of the data connection we used to prefill if we used one
+    // TODO this should be done as part of https://issues.redhat.com/browse/RHOAIENG-9914
```

Comment:
Looks like this ticket is closed. Is the comment still valid?

Reply:
Yeah, my bad, and it doesn't even apply since we are removing data connections